mk/opencl.mk

```diff
@@ -1,15 +1,9 @@
-# Default the path to the NVIDIA Run time Library
-OPENCL_LIB_PATH ?= /usr/local/cuda/lib64
-OPENCL_LIB_DIR := $(shell test -d $(OPENCL_LIB_PATH); echo $$?)
-
-ifneq ($(OPENCL_LIB_DIR),0)
-$(error "Please specify the path to OpenCL library.")
-endif
-
-OPENCL_LIB_AVAIL := $(shell ls $(OPENCL_LIB_PATH) | grep -oi libOpenCL.so > /dev/null; echo $$?)
+# The auto-detected path of the OpenCL library
+OPENCL_LIB_PATH := $(shell pkg-config --variable=libdir OpenCL 2> /dev/null)
```

Comment:
It is not macOS-friendly. Please check the following to understand how macOS works:
* [A Makefile for OpenCL development](http://cjlarose.com/2015/02/05/makefile-for-opencl-development.html)
* [Using OpenCL on Mac OS X - Test for OpenCL Devices](https://gist.github.com/pklaus/3858016)

Reply:
The `LDFLAGS` has been set properly.